docs/source/cpp/compute.rst

```diff
@@ -542,16 +542,17 @@ cast to the :ref:`common numeric type <common-numeric-type>` before comparison),
 or two inputs of Binary- or String-like types, or two inputs of Temporal types.
 If any input is dictionary encoded it will be expanded for the purposes of
 comparison. If any of the input elements in a pair is null, the corresponding
-output element is null.
-
-+--------------------------+------------+---------------------------------------------+---------------------+
-| Function names           | Arity      | Input types                                 | Output type         |
-+==========================+============+=============================================+=====================+
-| equal, not_equal         | Binary     | Numeric, Temporal, Binary- and String-like  | Boolean             |
-+--------------------------+------------+---------------------------------------------+---------------------+
-| greater, greater_equal,  | Binary     | Numeric, Temporal, Binary- and String-like  | Boolean             |
-| less, less_equal         |            |                                             |                     |
-+--------------------------+------------+---------------------------------------------+---------------------+
+output element is null. Decimal arguments will be promoted in the same way as
+for ``add`` and ``subtract``.
+
++--------------------------+------------+------------------------------------------------------+---------------------+
+| Function names           | Arity      | Input types                                          | Output type         |
++==========================+============+======================================================+=====================+
+| equal, not_equal         | Binary     | Numeric, Temporal, Decimal, Binary- and String-like  | Boolean             |
```

Comment:
Is `Decimal` not part of `Numeric`?  I don't see any reason to call it out special here given that we don't do so for add/multiply/divide/subtract.

Although, to be fair, we don't currently support decimal for the rest of the math operations (trig, pow/ln, etc.) and that's more than just a casting issue so it may be sometime before we do offer support.  So we might want to call out that fact.

Either way, I think we should keep this page consistent.  Either decimal is part of "Numeric" everywhere or nowhere.

Reply:
Fair, "Numeric" is defined up top as often including decimal. The blurb about promotion here should suffice, I think; I also added a blurb about decimal promotion to the arithmetic section.